ESQL: Add syntax clarifications on identifiers and literals (#101790)
elastic/elasticsearch: docs/reference/esql/esql-syntax.asciidoc

----
====

[discrete]
[[esql-identifiers]]
==== Identifiers

The identifiers can be used as they are and don't require quoting, unless
containing special characters, in which case they must be quoted with
backticks (+{backtick}+). What "special characters" means is command dependent.

For <<esql-from, FROM>>, <<esql-keep, KEEP>>, <<esql-drop, DROP>>,
<<esql-rename, RENAME>>, <<esql-mv_expand, MV_EXPAND>> and

> <<esql-enrich, ENRICH>> these are: `=`, ```, `,`, ` ` (space), `|` ,

Author: good catch, thanks.

----
====

[discrete]
[[esql-identifiers]]
==== Identifiers

> Unless containing special characters, the identifiers can be used as-are and

Reviewer: ```suggestion
Unless containing special characters, the identifiers can be used as is and
```

Author: Thanks.

don't require quoting. If they do, however, they must be quoted with

> backticks (```). The special characters are those that have already a semantic

Reviewer 2: ```suggestion
backticks (+{backtick}+). The special characters are those that have already a semantic
```
Getting a single backtick in an inline code block is not trivial.

Author: Nice, thanks!

meaning for a given command, like arithmetic operators (`-`, `+` etc.),
decimals delimiters (`.`), field delimiters (`,`), command separators (`|`),

> subfield referencing (`.`) a.s.o. This is also applicable to identifiers that

Reviewer: `a.s.o.`? Maybe provide all of them?

Author: I've reworked this paragraph to simplify it.

> could be interpreted like numbers (the scientific notation, like `1e2`) or

Reviewer: ```suggestion
could be interpreted as numbers (for example, the scientific notation `1e2`) or
```

simply start with a number.

For instance:

[source,esql]
----
// Retain just one field
FROM index
| KEEP 1.field
----

is legal. However, if same field is to be used with an <<esql-eval, EVAL>>,
it'd have to be quoted:

[source,esql]
----
// Copy one field
FROM index
| EVAL my_field = `1.field`
----

[discrete]
[[esql-literals]]
==== Literals

> {esql} supports currently numeric and string literals. The numeric literals

Reviewer: ```suggestion
{esql} currently supports numeric and string literals. The numeric literals
```

> can be used as are. The string literals need to be quoted in double

Reviewer: `as they are` or `as is`. I believe `as are` is incorrect.

Author: Reworked this paragraph too, splitting strings and numeric literals.
